frontend/src/modules/scaffold/views/scaffoldAssetItemView.js

```diff
@@ -0,0 +1,221 @@
+define([
+  'core/origin',
+  'backbone-forms',
+  'core/helpers',
+  './scaffoldAssetView',
+  'core/models/courseAssetModel',
+  'modules/assetManagement/views/assetManagementModalView',
+  'modules/assetManagement/collections/assetCollection'
+], function(Origin, BackboneForms, Helpers, ScaffoldAssetView, CourseAssetModel, AssetManagementModalView, AssetCollection) {
+
+  var ScaffoldAssetItemView = ScaffoldAssetView.extend({
+
+    events: function() {
+      return _.extend({}, ScaffoldAssetView.prototype.events, {
+        'click .add-item': 'onAddItemClick',
+        'click .ui-draggable': 'onDragItemClick'
+      });
+    },
+
+  	onAddItemClick: function(event) {
+      event.preventDefault();
+      this.itemListView.addItem();
+    },
+
+    onDragItemClick: function(event) {
+      var item = _.find(this.dragItems, function(i) {
+        return i.$el[0] === event.currentTarget;
+      });
+      item.view.openEditor();
+    },
+
+    initialize: function(options) {
+      ScaffoldAssetView.prototype.initialize.apply(this, arguments);
+      this.init(options);
+    },
+
+    init: function(options) {
+      var attrs = options.schema.inputType;
+      this.dragItems = [];
+      this.precision = attrs.precision || 3;
+      this.topAttr = attrs.topAttribute || '_top';
+      this.leftAttr = attrs.leftAttribute || '_left';
+      this.itemsPath = attrs.items || 'properties._items';
+      this.itemsPath = this.itemsPath.split('.');
+    },
+
+    postRender: function() {
+      if (!this.itemListView) {
+        this.itemListView = this.findItemView();
+        this.listenTo(this.itemListView, {
+          add: this.onItemListAdd,
+          remove: this.onItemListRemove,
+          change: this.onItemListChange
+        });
+      }
+      // double defer:
+      // if external asset, postRender is triggered before el is injected into Dom
+      _.defer(function() {
+        var items = this.itemListView.items;
+        this.removeDragItems();
+        items && items.forEach(function(item) {
+          this.addDragItem(item.editor);
+        }, this);
+      }.bind(this));
+    },
+
+    findItemView: function() {
+      var form = Origin.scaffold.getCurrentForm();
+
+      var targetAttr = this.schema.inputType.targetAttribute;
+      var itemAttr = this.schema.inputType.items;
+
+      var baseEditor;
+
+      switch (this.schema.inputType.type) {
+        case 'extension':
+          baseEditor = form.fields._extensions.editor.nestedForm.fields[targetAttr].editor;
+        case 'menu':
+          baseEditor = form.fields.menuSettings.editor.nestedForm.fields[targetAttr].editor;
+        default:
+          baseEditor = form.fields.properties.editor;
+      }
+
+      return baseEditor.nestedForm.fields[itemAttr].editor;
```

Comment:
This function is looking a lot better. A few things:
* We ought to be able to handle a schema with just `"inputType": "AssetItem"` since we've already gone some of the way by applying fallbacks in the `init` function
* `this.schema.inputType.type` needs a different name since `type` would be `AssetItem` now. Is there a way to test the extension/menu situations?
* I'd cache `this.schema.inputType` in any case
* Switch statement is missing its `break`s